smartsim/_core/_cli/test.py

```diff
@@ -0,0 +1,207 @@
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import io
+import tempfile
+import typing as t
+from contextlib import contextmanager
+import socket
+from types import TracebackType
+
+import numpy as np
+from smartredis import Client
+
+from smartsim import Experiment
+from smartsim._core.utils.helpers import installed_redisai_backends
+from smartsim._core._cli.utils import SMART_LOGGER_FORMAT
+from smartsim.log import get_logger
+
+SMART_LOGGER_FORMAT = "[%(name)s] %(levelname)s %(message)s"
+logger = get_logger("Smart", fmt=SMART_LOGGER_FORMAT)
+
+# Many of the functions in this module will import optional
+# ML python packages only if they are needed to test the build is working
+#
+# pylint: disable=import-error,import-outside-toplevel
+# mypy: disable-error-code="import"
+
+
+class _VerificationTempDir(tempfile.TemporaryDirectory):
+    """A Temporary directory to be used as a context manager that will only
+    clean itself up if no error is raised within its context
+    """
+
+    def __exit__(
+        self,
+        exc: t.Optional[t.Type[BaseException]],
+        value: t.Optional[BaseException],
+        tb: t.Optional[TracebackType],
+    ) -> None:
+        if not value:  # Yay, no error! Clean up as normal
+            super().__exit__(exc, value, tb)
+        else:  # Uh-oh! Better make sure this is not implicitly cleaned up
+            self._finalizer.detach()  # type: ignore[attr-defined]
+
+
+def execute(_args: argparse.Namespace, /) -> int:
+    backends = installed_redisai_backends()
+    try:
+        with _VerificationTempDir() as temp_dir:
+            test_install(
+                location=temp_dir,
+                port=None,  # TODO: allow users to pass as arg `--port`?
```

Comment:
I'll steal from Matt and ask if you want to create a task for this and remove the TODO from the code

Reply:
damn it, the consequences of my actions are catching up with me :(. Consider it made if/when this goes in!

Reply:
Other additional arguments were requested for the `validate` target. A `--port` was additionally added in d9316f595f099a8ee79c0dd042bdca8acaa7532d